metadata/src/main/java/org/apache/kafka/image/loader/MetadataLoader.java

```diff
@@ -197,13 +209,15 @@ private MetadataLoader(
         String threadNamePrefix,
         FaultHandler faultHandler,
         MetadataLoaderMetrics metrics,
-        Supplier<OptionalLong> highWaterMarkAccessor
+        Supplier<OptionalLong> highWaterMarkAccessor,
+        Supplier<LeaderAndEpoch> leaderAndEpochAccessor
     ) {
         this.log = logContext.logger(MetadataLoader.class);
         this.time = time;
         this.faultHandler = faultHandler;
         this.metrics = metrics;
         this.highWaterMarkAccessor = highWaterMarkAccessor;
+        this.leaderAndEpochAccessor = leaderAndEpochAccessor;
```

Comment:
Should we call this "initialLeaderAndEpochAccessor" since we only read from it during startup? Otherwise it might be confusing that we have this supplier, but update the `currentLeaderAndEpoch` in a different way